Projects/UOContent/Misc/CharacterCreation.cs

```diff
@@ -76,8 +76,111 @@ public static class CharacterCreation
+        new("Britain", "Sweet Dreams Inn", 1496, 1628, 10, Map.Felucca),
+        // TODO: Add New Magincia
+        new("Moonglow", "The Scholars Inn", 4408, 1168, 0, Map.Felucca),
+        new("Trinsic", "The Traveler's Inn", 1845, 2745, 0, Map.Felucca),
+        new("Magincia", "The Great Horns Tavern", 3734, 2222, 20, Map.Felucca),
+        new("Jhelom", "The Mercenary Inn", 1374, 3826, 0, Map.Felucca),
+        new("Skara Brae", "The Falconer's Inn", 618, 2234, 0, Map.Felucca),
+        new("Vesper", "The Ironwood Inn", 2771, 976, 0, Map.Felucca),
+    };
+
+    // TODO: Move to JSON files
+    public static readonly CityInfo[] TrammelStartingCities =
+    {
+        new("New Haven", "New Haven Bank", 1150168, 3667, 2625, 0, Map.Trammel),
+        new("Yew", "The Empath Abbey", 1075072, 633, 858, 0, Map.Trammel),
+        new("Minoc", "The Barnacle", 1075073, 2476, 413, 15, Map.Trammel),
+        new("Britain", "The Wayfarer's Inn", 1075074, 1602, 1591, 20, Map.Trammel),
+        // TODO: Add New Magincia
+        new("Moonglow", "The Scholars Inn", 1075075, 4408, 1168, 0, Map.Trammel),
+        new("Trinsic", "The Traveler's Inn", 1075076, 1845, 2745, 0, Map.Trammel),
+        new("Jhelom", "The Mercenary Inn", 1075078, 1374, 3826, 0, Map.Trammel),
+        new("Skara Brae", "The Falconer's Inn", 1075079, 618, 2234, 0, Map.Trammel),
+        new("Vesper", "The Ironwood Inn", 1075080, 2771, 976, 0, Map.Trammel)
+    };
+
+
+    private static CityInfo[] _availableStartingCities;
+
+    public static CityInfo[] GetStartingCities(bool isYoung)
+    {
+        if (isYoung && ExpansionInfo.CoreExpansion.MapSelectionFlags.Includes(MapSelectionFlags.Trammel))
+        {
+            return TileMatrix.Pre6000ClientSupport ? OldHavenBank : NewHavenInn;
+        }
+
+        return _availableStartingCities ??= ConstructAvailableStartingCities();
+    }
+
+    private static CityInfo[] ConstructAvailableStartingCities()
+    {
+        if (!TileMatrix.Pre6000ClientSupport)
+        {
+            return OldHavenStartingCities;
+        }
+
+        var availableMaps = ExpansionInfo.CoreExpansion.MapSelectionFlags;
+        var trammelAvailable = availableMaps.Includes(MapSelectionFlags.Trammel);
+        var feluccaAvailable = availableMaps.Includes(MapSelectionFlags.Felucca);
+
+        var length = (trammelAvailable ? TrammelStartingCities.Length : 0) +
+                     (feluccaAvailable ? FeluccaStartingCities.Length : 0);
+
+        var cities = new CityInfo[length];
+        var index = 0;
+        if (trammelAvailable)
+        {
+            Array.Copy(TrammelStartingCities, 0, cities, index, TrammelStartingCities.Length);
+            index += TrammelStartingCities.Length;
+        }
+
+        if (feluccaAvailable)
+        {
+            Array.Copy(FeluccaStartingCities, 0, cities, index, FeluccaStartingCities.Length);
+        }
+
+        // TODO: Add Royal City for gargoyles
+
+        if (cities.Length == 0)
```

Comment:
Should this be moved up to line 161, and maybe have a return statement if `cities.length == 0`?  Just wondering if there are no cities available, do we need to do the Array.Copy?  Also, will it cause issues if (somehow) TrammelStartingCities.length? == 0 AND FeluccaStartingCities.Length == 0, and we try to do the Array.Copy??

Reply:
Yeah good call! - And yeah, it will cause issues, hence the warning. Specifically, players won't be able to log in. We don't want to send them to a map that isn't available, for whatever reason. So we either have to crash, or throw a warning/error.

Thoughts on other options?

Reply:
ahh ok.  Yeah, I can't really think of any other good way to handle that.  Like you said, throwing a warning/error is probably the best bet